symcrypt/src/main.rs

```diff
@@ -0,0 +1,31 @@
+use symcrypt::*;
+
+fn main() {
+    println!("Testing sha256! \n\n");
+
+    let data = b"this is a test";
+    let mut result: [u8; SHA256_RESULT_SIZE] = [0; SHA256_RESULT_SIZE];
+    sha256(data, &mut result);
+    println!("SHA256 stateless hash: {:?}", result);
+
+    let mut sha_test = Sha256State::new();
+
+    let mut result: [u8; SHA256_RESULT_SIZE] = [0; SHA256_RESULT_SIZE];
+    Sha256State::append(&mut sha_test, b"this is a test");
+    Sha256State::result(&mut sha_test, &mut result);
+    println!("SHA256 state hash: {:?}", result);
```

Comment:
>println!("SHA256 state hash: {:?}", result); [](http://example.com/codeflow?start=4&length=44)

Suggest comparing the result against a known-correct value. Same for SHA 384.

Reply:
Good Idea, is there a value that you suggest?

Reply:
If this is a test program, it might be worth ingesting some other Rust crypto library and comparing outputs.

You can also hardcode the output you have got from the test program so far back into the test program, and double check with an online hash generator / python / whatever.
i.e. I believe the result with SHA256 for your current data should be: 2e99758548972a8e8822ad47fa1017ff72f06f3ff6a016851f45c398732bc50c

Reply:
I have checked the test outputs with the online generators and can confirm that it is the same. I will eventually move this testing functionality over to a test file where we can assert against the hardcoded known values ex: 2e99758548972a8e8822ad47fa1017ff72f06f3ff6a016851f45c398732bc50c